ns1/config.go

```diff
@@ -37,10 +45,69 @@ func (c *Config) Client() (*ns1.Client, error) {
 		httpClient.Transport = tr
 	}
 
-	client := ns1.NewClient(httpClient, decos...)
+	// If NS1_DEBUG is set, define custom Doer to log HTTP requests made by SDK
+	if os.Getenv("NS1_DEBUG") != "" {
+		doer := ns1.Decorate(httpClient, Logging())
+		client = ns1.NewClient(doer, decos...)
+	} else {
```

Comment:
shadowing is OK, right? Could we just `httpClient := ns1.Decorate(httpClient, Logging())` and then get rid of the else?

Reply:
ah, got it, it's a type change